clients/drcachesim/tools/filter/record_filter.cpp

```diff
@@ -128,23 +143,75 @@ record_filter_t::parallel_shard_error(void *shard_data)
     return per_shard->error;
 }
 
+bool
+record_filter_t::write_trace_entry(per_shard_t *shard, const trace_entry_t &entry)
+{
+    if (!shard->writer->write((char *)&entry, sizeof(entry))) {
+        shard->error = "Failed to write to output file " + shard->output_path;
+        success_ = false;
+        return false;
+    }
+    ++shard->output_entry_count;
+    return true;
+}
+
 bool
 record_filter_t::parallel_shard_memref(void *shard_data, const trace_entry_t &entry)
 {
     per_shard_t *per_shard = reinterpret_cast<per_shard_t *>(shard_data);
+    ++per_shard->input_entry_count;
     bool output = true;
-    for (int i = 0; i < (int)filters_.size(); ++i) {
-        if (!filters_[i]->parallel_shard_filter(entry, filter_shard_data_[i]))
+    if (stop_timestamp_us_ == 0 ||
```

Comment:
What does per_shard->enabled mean?  I would expect it to guard all the code here and just exit up front if not enabled.

Reply:
Hmm.. yeah my code here was a bit convoluted wrt the `enabled` var. Fixed it -- used `enabled` to guard the parallel_shard_filter calls.. Though we cannot just exit early from this routine.